server/src/main/java/org/elasticsearch/index/mapper/DateFieldMapper.java

```diff
@@ -913,6 +914,17 @@ protected String contentType() {
 
     @Override
     protected void parseCreateField(DocumentParserContext context) throws IOException {
+        if (context.parser().currentToken() == XContentParser.Token.START_OBJECT) {
```

Comment:
Maybe we should check for start array? Or maybe even better check for `context.parser().currentToken().isValue() == false` here?

Reply:
I tested arrays too and they look ok.

Reply:
Ah, right arrays are valid. So we should allow that. In fact start_array is handled elsewhere.

Reply:
`isValue` is a bit problematic because it returns `false` for `null` and that's not what we want